libpod/container_api.go

```diff
@@ -478,13 +479,15 @@ func (c *Container) RemoveArtifact(name string) error {
 }
 
 // Wait blocks until the container exits and returns its exit code.
-func (c *Container) Wait() (int32, error) {
-	return c.WaitWithInterval(DefaultWaitInterval)
+func (c *Container) Wait(ctx context.Context) (int32, error) {
+	return c.WaitWithInterval(ctx, DefaultWaitInterval)
 }
 
+var errWaitingCanceled = errors.New("waiting was canceled")
```

Comment:
Might be worth making this a bit more generic (`ErrCanceled = errors.New("cancelled by user")`) and putting it in `define/errors.go` for other parts of Podman to use

Reply:
or maybe better yet use `context.Canceled`